src/fold/transformations/features.py

```diff
@@ -102,8 +109,19 @@ def transform(self, X: pd.DataFrame, in_sample: bool) -> pd.DataFrame:
 
         elif self.type is LabelingMethod.weekday_weekend_uniqueholiday:
             extra_holiday_features = get_multi_holidays(
-                X.index, self.country_codes
-            ).add(get_weekends(X.index), axis="index")
+                X.index, self.country_codes, encode=self.encode_holidays
+            )
+
+            extra_holiday_features.apply(
```

Comment:
`.apply` is inefficent compared to vectorized operations, is there any way to transform it into that?